docs/functionality/job-level-affinity/job-level-affinity-overview.md

```diff
@@ -0,0 +1,251 @@
+
+### Key Features of Job Level Affinity
+
+1. **Topology Domain Scheduling**: 
+   - PodGroups are scheduled within specific topology domains that have hierarchical relationships. This is configured through the `PodGroupAffinity` field, similar to traditional Pod or Node affinity.
+
+2. **Preferred and Required Affinity Terms**: 
+   - These terms define the conditions for PodGroup scheduling.
+   - **Required**: Specifies that Pods must be scheduled within the same network segment.
+   - **Preferred**: A softer constraint, which divides nodes based on topology keys but allows for more flexibility in scheduling across different values within that topology.
+   The 'preferred' array is ordered according to the actual topology layers from the largest to the smallest scale.
+   This ordering is crucial for ensuring that the scheduler considers the topology layers in the correct hierarchy.
+
+3. **Node Selector Integration**:
+   - Incorporates node selectors in the PodGroup configuration to ensure Pods are scheduled on nodes meeting specific criteria, optimizing the scheduling process by pre-filtering nodes based on these requirements.
+
+4. **Enhanced NodeCircle Sorting**:
+   - When dividing NodeGroups based on affinity, the scheduler also creates NodeCircles within these NodeGroups, representing different levels within the network topology.
+   Sorting of NodeGroups and NodeCircles is based on configurable strategies, considering either available or total resources (CPU, MEM, GPU) in ascending or descending order. 
+   This approach enhances scheduling efficiency and minimizes resource fragmentation.
+
+This introduction provides a fundamental understanding of Job Level Affinity in the Godel Scheduler, highlighting its role in optimizing Pod scheduling in accordance with network topology requirements.
+
+
+## How to Use Job Level Affinity
+
+Using Job Level Affinity in the Godel Scheduler involves configuring the `affinity` field in the PodGroup specification.
+This configuration controls how PodGroups are scheduled within the cluster, based on their network topology and other criteria. 
+
+### Setting Up Affinity in PodGroup Spec
+
+1. **Configure Preferred and Required Affinity**:
+   - **Preferred Affinity**: Indicates an attempt to schedule Pods in a specific domain within a topology dimension.
+   If not feasible, it degrades to a higher-level domain. 
+   At the same time, it aims to minimize crossings in the next lower-level topology domains to avoid random placement across the higher-level topology domain.
+   - **Required Affinity**: Specifies that Pods must be scheduled within a specific domain of a topology dimension.
+
+   Example:
+
+   ```yaml
+   apiVersion: scheduling.orchestration.bytedance.com/v1alpha1
+   kind: PodGroup
+   metadata:
+     generation: 1
+     name: nginx
+   spec:
+     affinity:
+       podGroupAffinity:
+         preferred:
+         - topologyKey: tce.kubernetes.io/nettoripv6
```

Comment:
Is it ok to use 'tce' for open-source docs? Shall we change the key name to a more general one?

Reply:
done